lib/units/storage/amazons3.js

```diff
@@ -0,0 +1,197 @@
+var http = require('http')
+var util = require('util')
+var path = require('path')
+
+var express = require('express')
+var validator = require('express-validator')
+var bodyParser = require('body-parser')
+var formidable = require('formidable')
+var Promise = require('bluebird')
+var uuid = require('node-uuid')
+var fs = require('fs')
+
+var lifecycle = require('../../util/lifecycle')
+var logger = require('../../util/logger')
+var requtil = require('../../util/requtil')
+
+var AWS = require('aws-sdk')
+
+module.exports = function(options) {
+  var log = logger.createLogger('storage:s3')
+  , app = express()
+  , server = http.createServer(app)
+  , credentials = new AWS.SharedIniFileCredentials({
+      profile: options.profile
+    })
+  , bucket = options.Bucket
```

Comment:
I would get rid of `bucket` here, and just use options.bucket (lowercase) everywhere. Makes it more obvious where the value comes from.